android/app/src/main/java/com/dismoi/scout/accessibility/BackgroundService.kt

```diff
@@ -1,193 +1,229 @@
+
+    try {
+      val currentUrl = activeBrowserConfig.getCurrentUrlIn(root)
+
+      if (currentUrl == null) return
+
+      Log.d(TAG, "current URL is $currentUrl")
+
+      val matchingContexts = getContextsMatchingUrl(currentUrl)
+
+      Log.d(TAG, "found ${matchingContexts.size} contexts matching current url")
+
+      if (matchingContexts.size > 0) {
+
+        val noticesId =
+          matchingContexts
+            .filter { context -> !context.has("xpath") || context.isNull("xpath") } // TODO Use domain fields instead of xpath
+            .map { context -> context.getInt("noticeId") }
+            .distinct()
+
+        if (noticesId.size == lastNotices.size && noticesId.containsAll(lastNotices)) {
           return
         }
-      }
 
-      if (isLauncherActivated(packageName)) {
-        _hide = "true"
-        _packageName = packageName
-        handler.post(runnableCode)
-        return
+        show(noticesId)
+      } else {
+        hide()
       }
+    } catch (e: NoUrlInBrowserException)
+    {
+      // When page is scrolled, the URL is hidden, but we may still be on the same page …
+    }
 
-      val parentNodeInfo: AccessibilityNodeInfo = event.source ?: return
-
-      chrome.parentNodeInfo = parentNodeInfo
-      chrome._packageName = packageName
-
-      if (chrome.checkIfChrome()) {
-        if (
-          getEventType(event) == "TYPE_WINDOW_STATE_CHANGED" ||
-          getEventType(event) == "TYPE_WINDOW_CONTENT_CHANGED"
-        ) {
-          chrome.captureUrl()
-          if (chrome.chromeSearchBarEditingIsActivated()) {
-            _hide = "true"
-            handler.post(runnableCode)
-            return
-          }
-        }
-        if (getEventType(event) == "TYPE_VIEW_ACCESSIBILITY_FOCUSED") {
-          _eventTime = event.eventTime.toString()
-          _hide = "false"
-          _packageName = packageName
-          handler.post(runnableCode)
-        }
+    // Just an example of code of how to search for a product name
+//    val webview = Helpers.findWebview(root)
+//    if (webview != null) {
+//      Amazon.getProductTitle(webview)
+//    }
+  }
 
-        parentNodeInfo.recycle()
-      }
+  private fun hide() {
+    if (bound) {
+      lastNotices = listOf()
+      floatingService!!.hide()
+    }
 
-      return
+  }
+
+  @RequiresApi(Build.VERSION_CODES.N)
+  private fun show(noticesIds: List<Int>) {
+    if (bound) {
+      lastNotices = noticesIds
+      floatingService!!.showNotices(noticesIds)
     }
   }
 
-  override fun onInterrupt() {
-    sendEventFromAccessibilityServicePermission("false")
+  // TODO To be moved to repository
+  private fun fetchMatchingContexts() {
+    val matchingContextsEndpoint = "https://notices.bulles.fr/api/v3/matching-contexts" // TODO to be moved to env configuration
+    val request = Request.Builder().url(matchingContextsEndpoint).build()
+    val client = OkHttpClient()
```

Comment:
Yes, I wanted to not put everything in this single PR … This will totally be refactored as part of https://github.com/dis-moi/mobile/issues/173#issuecomment-886119063